contracts/fees_collector_contract.py

```diff
@@ -188,22 +288,68 @@ def set_locked_tokens_per_block(self, deployer: Account, proxy: ProxyNetworkProv
         ]
         return endpoint_call(proxy, gas_limit, deployer, Address(self.address), "setLockedTokensPerBlock", sc_args)
 
+    def set_base_token_burn_percent(self, deployer: Account, proxy: ProxyNetworkProvider, base_token_burn_percentage: int):
+        function_purpose = f"Set base token burn percentage"
+        logger.info(function_purpose)
+
+        gas_limit = 30000000
+        sc_args = [
+            base_token_burn_percentage
+        ]
+        return endpoint_call(proxy, gas_limit, deployer, Address(self.address), "setBaseTokenBurnPercent", sc_args)
+
     def claim_rewards(self, user: Account, proxy: ProxyNetworkProvider):
         function_purpose = f"Claim rewards from fees collector"
         logger.info(function_purpose)
 
-        gas_limit = 80000000
+        gas_limit = 350000000
         sc_args = []
         return endpoint_call(proxy, gas_limit, user, Address(self.address), "claimRewards", sc_args)
     
     def claim_boosted_rewards(self, user: Account, proxy: ProxyNetworkProvider):
         function_purpose = f"Claim rewards from fees collector"
         logger.info(function_purpose)
 
-        gas_limit = 80000000
+        gas_limit = 500000000
         sc_args = []
         return endpoint_call(proxy, gas_limit, user, Address(self.address), "claimBoostedRewards", sc_args)
 
+    def redistribute_rewards(self, deployer: Account, proxy: ProxyNetworkProvider):
+        function_purpose = f"Redistribute rewards from fees collector"
+        logger.info(function_purpose)
+
+        gas_limit = 80000000
+        sc_args = []
+        return endpoint_call(proxy, gas_limit, deployer, Address(self.address), "redistributeRewards", sc_args)
+    
+    def deposit_swap_fees(self, user: Account, proxy: ProxyNetworkProvider):
+        function_purpose = f"Deposit swap fees in fees collector"
+        logger.info(function_purpose)
+
+        gas_limit = 80000000
+        sc_args = [
+        ]
+        return endpoint_call(proxy, gas_limit, user, Address(self.address), "depositSwapFees", sc_args)
+
+    def swap_token(self, user: Account, proxy: ProxyNetworkProvider, token: str, pair_contract: PairContract):
```

Comment:
First, I would avoid passing the PairContract object as parameter, especially if you need only its address. You can make it a str and expect to pass the bech32 address.
Second, I would group the token and pair_contract params in a args list, to follow the general structure for endpoints. We'll probably want to refactor this args list at some point (maybe into a **kwargs type) and it would be nice to have them all fall under a similar category for easier changing.